metricq_manager/config_parser.py

```diff
@@ -0,0 +1,352 @@
+                        self.client_token,
+                        deprecated_key,
+                        self.top_level_key,
+                        key,
+                    )
+                    return value
+
+        top_level: Optional[ClientConfig] = self.config.get(self.top_level_key)
+        if top_level is not None:
+            return top_level.get(key, default)
+        else:
+            return default
+
+    def classic_arguments(self) -> Iterator[Tuple[str, Any]]:
+        """An iterator over `key-value` pairs of arguments for queues of type
+        :literal:`"classic"`, as parsed from the configuration object.
+        """
+        if (message_ttl := self.message_ttl()) is not None:
+            yield ("x-message-ttl", message_ttl)
+
+    def message_ttl(self) -> Optional[int]:
+        """Parse message `time-to-live <https://www.rabbitmq.com/ttl.html#queue-ttl>`_
+        argument for messages in a classic queue.
+
+        Does not apply to quorum queues.
+        """
+        message_ttl: Any = self.get(
+            f"{self.role}-message-ttl", deprecated=["message_ttl"]
+        )
+
+        if message_ttl is None:
+            return None
+        elif isinstance(message_ttl, (float, int)):
+            ttl = int(1000 * message_ttl)
+
+            if ttl >= 0:
+                return ttl
+            else:
+                logger.warning(
+                    "Client {!r} has per-queue message TTL that is not a positive number of seconds (got {})",
+                    self.client_token,
+                    ttl,
+                )
+                return None
+        else:
+            logger.warning(
+                "Client {!r} has message TTL set which is not a number of seconds: got {} of type {!r}",
+                self.client_token,
+                message_ttl,
+                type(message_ttl),
+            )
+            return None
+
+    def quorum_arguments(self) -> Iterator[Tuple[str, Any]]:
+        """An iterator over `key-value` pairs of arguments for queues of type
+        :literal:`"quorum"`, as parsed from the configuration object.
+        """
+        yield ("x-queue-type", "quorum")
+
+    def arguments(self) -> Iterator[Tuple[str, Any]]:
+        """An iterator over `key-value` pairs of arguments for queues, as
+        parsed from the configuration object.
+
+        Depending on the queue type set in :literal:`{top_level_key}.{role}-queue-type`,
+        this yields either "classic" queue arguments
+        (:meth:`classic_arguments`) or "quorum" queue arguments
+        (:meth:`quorum_arguments`).
+        """
+        queue_type = self.queue_type()
+        if queue_type is QueueType.CLASSIC:
+            return self.classic_arguments()
+        elif queue_type is QueueType.QUORUM:
+            return self.quorum_arguments()
+        else:
+            assert False, f"Unhandled queue type: {queue_type!r}"
```

Comment:
Like [here](https://github.com/metricq/metricq-manager/pull/19#discussion_r571462159), I'd argue that asserting this is fine, since `queue_type` should never be anything other than `CLASSIC` or `QUORUM`.  Python does not have exhaustive enum checks, but a good type checker [might catch](https://github.com/python/mypy/issues/6366#issuecomment-560369716) that someone forgot to match a enum variant.